app/frontend/src/features/auth/email/ConfirmChangeEmail.tsx

```diff
@@ -18,25 +20,39 @@ export default function ConfirmChangeEmail() {
   const { resetToken } = useParams<{ resetToken?: string }>();
 
   const {
+    data,
     error,
     isLoading,
     isSuccess,
-    mutate: completeChangeEmail,
-  } = useMutation<Empty, GrpcError, string>((resetToken) =>
-    service.account.completeChangeEmail(resetToken)
+    mutate: confirmChangeEmail,
+  } = useMutation<ConfirmChangeEmailRes.AsObject, GrpcError, string>(
+    (resetToken) => service.account.confirmChangeEmail(resetToken)
   );
 
   useEffect(() => {
     if (resetToken) {
-      completeChangeEmail(resetToken);
+      confirmChangeEmail(resetToken);
     }
-  }, [completeChangeEmail, resetToken]);
+  }, [confirmChangeEmail, resetToken]);
+
+  function successMsg(state: EmailConfirmationState) {
+    switch (state) {
+      case EmailConfirmationState.EMAIL_CONFIRMATION_STATE_SUCCESS:
+        return CHANGE_EMAIL_SUCCESS;
+      case EmailConfirmationState.EMAIL_CONFIRMATION_STATE_REQUIRES_CONFIRMATION_FROM_NEW_EMAIL:
+        return CHANGE_EMAIL_NEED_NEW;
+      case EmailConfirmationState.EMAIL_CONFIRMATION_STATE_REQUIRES_CONFIRMATION_FROM_OLD_EMAIL:
+        return CHANGE_EMAIL_NEED_OLD;
+      default:
```

Comment:
Does the backend ever return `UNKNOWN` in practice (which is when this code path is hit here)? Maybe worth throwing an error here if it does as it'd be exceptional/unexpected and it shouldn't be a success message?

Reply:
Yeah, just with `throw Error("Invalid ...")`?

Reply:
Yup 👍

Reply:
This is only testable locally I assume since the staging API is not updated until this is merged?

Reply:
Yeah that's right